x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/buttons/dimension_button.tsx

```diff
@@ -25,33 +27,47 @@ export function DimensionButton({
   onRemoveClick,
   accessorConfig,
   label,
-  invalid,
+  message,
 }: {
   group: VisualizationDimensionGroupConfig;
   children: React.ReactElement;
   onClick: (id: string) => void;
   onRemoveClick: (id: string) => void;
   accessorConfig: AccessorConfig;
   label: string;
-  invalid?: boolean;
+  message: UserMessage | undefined;
 }) {
   return (
     <>
-      <EuiLink
-        className="lnsLayerPanel__dimensionLink"
-        data-test-subj="lnsLayerPanel-dimensionLink"
-        onClick={() => onClick(accessorConfig.columnId)}
-        aria-label={triggerLinkA11yText(label)}
-        title={triggerLinkA11yText(label)}
-        color={invalid ? 'danger' : undefined}
-      >
-        <ColorIndicator accessorConfig={accessorConfig}>{children}</ColorIndicator>
-      </EuiLink>
+      <EuiFlexItem>
+        <EuiToolTip
+          content={message?.shortMessage || message?.longMessage || undefined}
+          position="left"
+        >
+          <EuiLink
+            className="lnsLayerPanel__dimensionLink"
+            data-test-subj="lnsLayerPanel-dimensionLink"
+            onClick={() => onClick(accessorConfig.columnId)}
+            aria-label={triggerLinkA11yText(label)}
+            title={triggerLinkA11yText(label)}
+            color={
+              message?.severity === 'error'
+                ? 'danger'
+                : message?.severity === 'warning'
+                ? 'warning'
+                : undefined
+            }
+          >
+            <DimensionButtonIcon message={message} accessorConfig={accessorConfig}>
+              {children}
+            </DimensionButtonIcon>
+          </EuiLink>
+        </EuiToolTip>
+      </EuiFlexItem>
       <EuiButtonIcon
         className="lnsLayerPanel__dimensionRemove"
         data-test-subj="indexPattern-dimension-remove"
-        iconType="cross"
-        iconSize="s"
+        iconType="trash"
         size="s"
```

Comment:
Possible to reduce the size here to `xs` to fit more comfortably in the dimension item box?

```suggestion
        size="xs"
```